semantics/cpp14/language/translation/name.k

```diff
@@ -268,16 +304,28 @@ module CPP-TRANSLATION-NAME
                     | resolveElabSpecifier(K, StrictList) [strict(1, 2), klabel(resolveElabSpecifier2)]
                     | specializeTypeTemplate(QualId, CPPType, List)
 
-     rule <k> nameLookup(X::CId, Tag::Tag, IsNNS::Bool) => lookupNameInFullClass(X, classFromScope(Scope), isBlockScope(Scope), Tag, IsNNS) ...</k>
+     rule <k> nameLookup(X::CId, Tag::Tag, IsNNS::Bool) => nameLookupInScope(X, Tag, IsNNS, Scope) ...</k>
           <curr-scope> Scope::Scope </curr-scope>
+
+     rule <k> nameLookupInScope(X::CId, Tag::Tag, IsNNS::Bool, Scope::Scope) => lookupNameInFullClass(X, classFromScope(Scope), isBlockScope(Scope), Tag, IsNNS) ...</k>
           requires inClassScope(Scope)
 
+     rule nameLookupInScope(X::CId, Tag::Tag, IsNNS::Bool, enumScope(E::EnumId)) => nameLookupInEnumScope(X, Tag, IsNNS, enumScope(E))
+
```

Comment:
And this might be problematic if you have an enum in a class, as in the next example, as if the first rule is chosen, the name will be looked up outside the class.
```
constexpr int a = 4;
struct C {
  enum Enum {
    a = 5,
    b = a*2
  };
};

int main() {
  C::Enum e = C::b;
  return e - 10;
}
```